### ISSUE TITLE
SC2154 applied

### ISSUE THREAD

**Issue by GregThib** [koalaman/shellcheck#1729]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

echo " - DO_JOBS   = ${DO_JOBS=yes}"
echo " - Timeout   = ${timeout=900}"

```

#### Here's what shellcheck currently says:
```
Line 4	SC2154: timeout is referenced but not assigned.
```
#### Here's what I wanted or expected to see:

No warning must be provided for the second line.
It's very surprising that the first line didn't throw any warning, but the second one yes.

**Comment by matthewpersico** [comment]
ALLCAPS variables are assumed to be defined and exported, so shellcheck does not complain about no previous encounter. Everything else is assumed to be script local so shellcheck will throw 2154 for the second one.

**Comment by GregThib** [comment]
Yeah, but… the second one reference and ASSIGN the variable. And the message says "not assigned".

(thank you for the tips about CAPS variables, i didn't knew that)

**Comment by matthewpersico** [comment]
Ah! I see. You believe the default assignment if not assigned should count as an assignment. Hmm. I think you have a point. Nice catch.

**Comment by xPMo** [comment]
Just came across this because I found it myself. I agree that 2154 should not apply to `${unset=value}`.
